### ISSUE TITLE
Feature Request: Add "Support end date" metadata

### ISSUE THREAD

**Issue by AndriySvyryd** [NuGet/Home#13774]
### NuGet Product(s) Involved

NuGet.exe, MSBuild.exe, dotnet.exe, NuGet SDK

### The Elevator Pitch

Packages that follow a set release lifecycle would expose this information. Then when building a package that claims to be supported longer than some of its transient dependencies NuGet would show a warning. This would help to proactively identify dependencies that would be deprecated before the using package.

### Additional Context and Details

An extra feature would be to allow extending the "Support end date" without releasing a new package version or reuploading it.

**Comment by kartheekp-ms** [comment]
I’m sure you are aware that nuget.org provides an option to [deprecate packages](https://learn.microsoft.com/nuget/nuget-org/deprecate-packages). Customers will be notified if they have installed deprecated packages in the Visual Studio Package Manager UI and also via the `dotnet list package --deprecated` command in the CLI.

**Comment by AndriySvyryd** [comment]
> I’m sure you are aware that nuget.org provides an option to deprecate packages. Customers will be notified if they have installed deprecated packages in the Visual Studio Package Manager UI and also via the dotnet list package --deprecated command in the CLI.

Right, but there's currently no way of knowing through NuGet when the packages will be deprecated. This is important information to know when choosing dependencies as the latest version isn't the one that will be supported for longer for an STS release.
